openwisp_monitoring/device/api/views.py

```diff
@@ -75,6 +76,9 @@ def get(self, request, pk):
             response = HttpResponse(self._get_csv(data), content_type='text/csv')
             response['Content-Disposition'] = 'attachment; filename=data.csv'
             return response
+        device_metrics_received.send(
```

Comment:
The metrics are not received in the GET, but in the POST: GET is used by the web UI to retrieve the data for the graphs, POST is used by the devices to send their metrics.
I didn't catch this earlier.

Please move the signal to the `post` method and update the tests accordingly.